Bio/Entrez/Parser.py

```diff
@@ -535,14 +514,15 @@ def count(model):
             self.structures.update({name: multiple})
 
     def open_dtd_file(self, filename):
-        path = os.path.join(DataHandler.local_dtd_dir, filename)
+        self._initialize_directory()
```

Comment:
Actually why was this line needed here? We are already calling `self._initialize_directory` from `directory.setter`.

Same for the calls to `self._initialize_directory` below.

Reply:
@mdehoon Great question. I'm refreshing my memory. 

Am I correct that the `directory.setter` function will be called:
1. when the DataHandler object is created (__init__)
2. again if the user sets a custom directory (local cache)
?

If that's true, then I see your point. There shouldn't be a need to call _initialize_directory yet again in the open* and save* functions, if self_directory should not have changed in the interim? It will already have gotten initialized one way or another? It's possible those 4 instances were added for another reason I can't recall. I apologize, I haven't been in this codebase since the PR.

Reply:
It's easy to set it up like that. But should `directory` be a class attribute or an object attribute? Right now it is an object attribute, meaning that it has to be set separately each time a `DataHandler` is created.

Reply:
Sorry for the delay in responding, @mdehoon . 

Can you catch me up? It looks like the `_initialize_directory` method from this PR has since evolved into the `directory` method decorated by `@directory.setter`. But it sounds like we are still making this unnecessary call to that code (formerly in `self.initialize_directory`) in the 4 open and save methods? Can you trace the story for me? 

Apologies, I'm rusty from being away from this repo. I'm very happy to help clean up my PR! 

When I look at the latest Master, `open_dtd_file` and cousin methods don't seem to initialize anymore or set `directory` directly. From your comment I am guessing each method somehow invokes a `directory=`, etc, and I'm just missing it now. Thank you!